From da15bc14c8140e7d4cefc28f940c84159dccdf4a Mon Sep 17 00:00:00 2001
From: "Alejandro G. Castro" <alex@igalia.com>
Date: Mon, 19 Oct 2026 09:06:22 +0200
Subject: [PATCH] [Android] Build fix after 323956@main: [[clang::suppress]] on
 a function needs Clang 19
 https://bugs.webkit.org/show_bug.cgi?id=NEED_BUG_URL

Reviewed by NOBODY (OOPS!).

323956@main put SUPPRESS_MEMORY_UNSAFE_CAST on the defaulted operator<=>()
of Seconds. That macro is for statements and variables: Clang accepts
[[clang::suppress]] on a declaration only since version 19, so with the
Android NDK's Clang 18 every file that includes Seconds.h fails:

    Seconds.h:208:5: error: 'suppress' attribute only applies to variables and statements

Use IGNORE_CLANG_STATIC_ANALYZER_WARNINGS_ATTRIBUTE_ON_FUNCTION(), which is
empty before Clang 19, as StyleOpacity.h already does for the same kind of
operator.

* Source/WTF/wtf/Seconds.h:
---
 Source/WTF/wtf/Seconds.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/WTF/wtf/Seconds.h b/Source/WTF/wtf/Seconds.h
index bc7f0c0f3a6..15f93de5cf0 100644
--- a/Source/WTF/wtf/Seconds.h
+++ b/Source/WTF/wtf/Seconds.h
@@ -205,7 +205,7 @@ public:
     WTF_EXPORT_PRIVATE TimeWithDynamicClockType LEGACY_NODELETE operator-(const TimeWithDynamicClockType&) const;
     
     // FIXME: Remove this suppression once rdar://190195764 is fixed.
-    SUPPRESS_MEMORY_UNSAFE_CAST friend constexpr auto operator<=>(Seconds, Seconds) = default;
+    IGNORE_CLANG_STATIC_ANALYZER_WARNINGS_ATTRIBUTE_ON_FUNCTION("alpha.webkit.MemoryUnsafeCastChecker") friend constexpr auto operator<=>(Seconds, Seconds) = default;
     
     WTF_EXPORT_PRIVATE void dump(PrintStream&) const;
     
